pkg/network/common/egressip.go

```diff
@@ -299,6 +366,108 @@ func (eit *EgressIPTracker) UpdateHostSubnetEgress(hs *osdnv1.HostSubnet) {
 	eit.recordMetrics()
 }
 
+func (eit *EgressIPTracker) GetNodeNameByNodeIP(nodeIP string) string {
+	if node, exists := eit.nodesByNodeIP[nodeIP]; exists {
+		return node.nodeName
+	}
+	return ""
+}
+
+func (eit *EgressIPTracker) watchNodes(nodeInformer kcoreinformers.NodeInformer) {
+	funcs := InformerFuncs(&corev1.Node{}, eit.handleAddOrUpdateNode, nil)
+	eit.nodeInformer.Informer().AddEventHandler(funcs)
+}
+
+func (eit *EgressIPTracker) handleAddOrUpdateNode(obj, _ interface{}, eventType watch.EventType) {
+	eit.Lock()
+	defer eit.Unlock()
+
+	node := obj.(*corev1.Node)
+	klog.V(5).Infof("Watch %s event for Node %q", eventType, node.Name)
+
+	if err := eit.initNodeCapacity(node); err != nil {
+		klog.Errorf("Error initializing capacity for Node %q, err: %v", node.Name, err)
+	}
+}
+
+func (eit *EgressIPTracker) getNodeInternalIP(node *corev1.Node) string {
+	for _, nodeAddress := range node.Status.Addresses {
+		if nodeAddress.Type == corev1.NodeInternalIP && net.ParseIP(nodeAddress.Address).To4() != nil {
+			return nodeAddress.Address
+		}
+	}
+	return ""
+}
+
+// egressCIDROverlapsCloudSubnet checks that whatever egress CIDRs are specified
+// on the HostSubnet also are completely overlapped by the cloud network. A
+// failure to specify this correctly will lead to egress IP assignments by the
+// SDN which are not allowed by the cloud provider.
+func (eit *EgressIPTracker) egressCIDROverlapsCloudSubnet(hs *osdnv1.HostSubnet) error {
+	node, err := eit.nodeInformer.Lister().Get(hs.Host)
+	if err != nil {
+		return fmt.Errorf("error listing node, err: %v", err)
```

Comment:
If the node has been deleted but the `HostSubnet` still exists and someone modifies that, we might be logging errors in other places as well and possibly be logging more errors than we need to, but this seems like a sufficiently improbable edge case that I am personally fine with returning the error and logging that, more rather than less.